apps/laboratory/app/core/universal-provider/page.tsx

```diff
@@ -0,0 +1,372 @@
+      }
+    }
+
+    initialize()
+  }, [])
+
+  async function handleConnect() {
+    if (!provider || !appKit) {
+      return
+    }
+
+    try {
+      setIsLoading(true)
+      await appKit.open()
+      appKit.subscribeEvents(({ data }: any) => {
+        if (data.event === 'MODAL_CLOSE') {
+          setIsLoading(false)
+        }
+      })
+      await provider.connect({ optionalNamespaces: OPTIONAL_NAMESPACES })
+    } catch (error) {
+      console.error('Connection error:', error)
+      toast({
+        title: 'Connection Error',
+        description: error instanceof Error ? error.message : 'Failed to connect',
+        type: 'error'
+      })
+    } finally {
+      setIsLoading(false)
+    }
+  }
+
+  async function handleDisconnect() {
+    if (!provider) {
+      return
+    }
+
+    try {
+      await provider.disconnect()
+      setAccount(undefined)
+      setNetwork(undefined)
+      setBalance(undefined)
+      toast({
+        title: ConstantsUtil.DisconnectingSuccessToastTitle,
+        description: 'Successfully disconnected',
+        type: 'success'
+      })
+    } catch (error) {
+      console.error('Disconnect error:', error)
+      toast({
+        title: ConstantsUtil.DisconnectingFailedToastTitle,
+        description: error instanceof Error ? error.message : 'Failed to disconnect',
+        type: 'error'
+      })
+    }
+  }
+
+  function getPayload() {
+    if (!account || !network) {
+      return null
+    }
+
+    const map = {
+      solana: {
+        method: 'solana_signMessage',
+        params: {
+          message: base58.encode(new TextEncoder().encode('Hello Appkit!')),
+          pubkey: account
+        }
+      },
+      eip155: {
+        method: 'personal_sign',
+        params: ['Hello AppKit!', account]
+      },
+      bip122: {
+        method: 'signMessage',
+        params: {
+          message: 'Hello AppKit!',
+          account
+        }
+      }
+    }
+
+    const [namespace] = network.split(':')
+
+    return map[namespace as keyof typeof map]
+  }
+
+  async function handleSignMessage() {
+    if (!provider || !account) {
+      toast({
+        title: 'Error',
+        description: 'User is disconnected',
+        type: 'error'
+      })
+
+      return
+    }
+
+    try {
+      const payload = getPayload()
+
+      if (!payload) {
+        toast({
+          title: 'Error',
+          description: 'Chain not supported by laboratory',
+          type: 'error'
+        })
+
+        return
+      }
+
+      if (!network) {
+        toast({
+          title: 'Error',
+          description: 'Network not supported by laboratory',
+          type: 'error'
+        })
+
+        return
+      }
+
+      const signature = await provider.request(payload, network)
+
+      toast({
+        title: ConstantsUtil.SigningSucceededToastTitle,
+        description: 'Message signed successfully',
+        type: 'success'
+      })
+
+      console.log('Signature:', signature)
```

Comment:
We can pass signature to the toast description like others and remove log maybe?